core/src/main/java/de/danoeh/antennapod/core/receiver/MediaButtonReceiver.java

```diff
@@ -25,6 +29,16 @@ public void onReceive(Context context, Intent intent) {
 		if (intent == null || intent.getExtras() == null) {
 			return;
 		}
+
+		if(intent.getAction().equals(NOTIFY_PROGRESS_STRING_RECEIVER)){
+			Log.d(TAG, "onReceive: got it");
+			boolean showTimeLeft = UserPreferences.shouldShowRemainingTime();
+			showTimeLeft = !showTimeLeft;
+			UserPreferences.setShowRemainTimeSetting(showTimeLeft);
```

Comment:
Curious why you are toggling the preference here?

Reply:
Thanks for reviewing @tonytamsf!

To be honest, I am not sure if we should actually toggle the preference from the widget. I would rather keep the rather big touch area that it currently has than to squeeze a small button in there.

Reply:
@tonytamsf @ByteHamster Yes we can change preference from widget, here I am changing it from a broadcast reciever. The preference is being toggled to switch from duration time to remaining time and vice versa on clicking the time in widget.